### PR TITLE
Update the publish.yml template

### LINKED ISSUE #17: [Bug] Using deprecated input for docker/build-push-action@v1
**Describe the bug**
During testing of the course I noticed there was a warning annotation saying the workflow is using a deprecated input for `docker/build-push-action@v1`

**To Reproduce**
1. Get to step 4 in the course
2. Click on **Actions** in the top nav
3. Select the latest run of **Publish to docker**
4. Scroll down to Annotations
5. See warning

**Expected behavior**
A workflow run without warning/error annotations

**Screenshots**
Screen Shot 2023-02-15 at 3 14 49 PM

**Device information**
 - Type: MacBook Pro
 - OS: macOS Big Sur
 - Browser: Chrome
 - Version: Version 109.0.5414.87 (Official Build) (x86_64)

**Additional context**
Workflow should probably be updated to use `docker/build-push-action@v2`

### PR DESCRIPTION
**Pull request by sinsukehlab**
metadata-action

### Why:

Closes #17

### What's being changed:

- Bump actions/checkout to v3 (already done in 2459f17)
- Add a docker/metadata-action@v4 step
- Add a docker/login-action@v2 step
- Bump docker/build-push-action to v3

### Check off the following:

- [ ] For workflow changes, I have verified the Actions workflows function as expected.
- [ ] For content changes, I have reviewed the [style guide](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,14 +87,24 @@ We'll start by creating the workflow file to publish a Docker image to GitHub Pa
        steps:
          - uses: actions/checkout@v3
          # Add your test steps here if needed...
-         - name: Build container
-           uses: docker/build-push-action@v1
+         - name: Docker meta
```
Comment (chadfawcett) [inline README.md:90]: @sinsukehlab Can you explain why this part is needed? Rather than just updating `docker/build-push-action` to v3?

Comment (sinsukehlab) [reply README.md:90]: Certainly. The Boolean env `tag_with_sha` is abolished in docker/build-push-action@v2 or higher. I use docker/metadata-action to handle tags.
See: https://docs.docker.com/build/ci/github-actions/examples/#manage-tags-and-labels

**Comment by chadfawcett** [comment]
@heiskr can you give this a review?

**Review by heiskr — APPROVED** [review]
All looks gravy to me :) Thanks!

File: README.md (hunk 1 above)
Comment (sinsukehlab) [reply README.md:91]: The link is now https://docs.docker.com/build/ci/github-actions/manage-tags-labels/

**Pull request merged**